src/multiple_shooting_objective.jl

```diff
@@ -0,0 +1,56 @@
+export multiple_shooting_objective
+function multiple_shooting_objective(prob::DEProblem,alg,loss,init_N_params,regularization=nothing;mpg_autodiff = false,
+                              verbose_opt = false,prob_generator = problem_new_parameters,autodiff_prototype = mpg_autodiff ? zeros(init_N_params) : nothing,
+                              autodiff_chunk = mpg_autodiff ? ForwardDiff.Chunk(autodiff_prototype) : nothing,
+                              kwargs...)
+  cost_function = function (p)
+    N = length(p)-2
+    time_len = Int(floor(length(loss.t)/N))
+    time_dur = loss.t[1:time_len]
+    sol = []
+    loss_val = 0
+    for i in 1:2:N
+        tmp_prob = remake(prob;u0=p[i:i+1],p=[N+1:N+2])
+        if typeof(loss) <: Union{CostVData,L2Loss,LogLikeLoss}
+          # println("huihui1")
+          push!(sol,solve(tmp_prob,alg;saveat=time_dur,save_everystep=false,dense=false,kwargs...))
+          # println("huihui2")
+          if (i+1)*time_len < length(loss.t)
+            time_dur = loss.t[i*time_len:(i+1)*time_len]
```

Comment:
this should be before you use it in saveat?

Reply:
Have initialized before and then updating it after using it in `saveat`